### PR TITLE
[INLONG-8171][Manager] Optimize the way of creating LoadNodes to make it easy to expand and maintain

### LINKED ISSUE #8171: [Improve][Manager] Optimize the way of creating LoadNodes to make it easy to expand and maintain
### Description

Optimize the way of creating LoadNodes, create LoadNodeFactory to make it a single responsibility, easy to expand and maintain

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by chestnut-c**
- Fixes #8171

### Motivation

Optimize the way of creating LoadNodes, create LoadNodeFactory to make it a single responsibility, easy to expand and maintain.

### CODE REVIEW

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/load/HudiProvider.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.inlong.sort.protocol.node.load.HudiLoadNode;
+import org.apache.inlong.sort.protocol.transformation.FieldRelation;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The Provider for creating Hudi load nodes.
+ */
+public class HudiProvider implements LoadNodeProvider {
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/load/HudiProvider.java:37]: Can we merge the HudiProvider into one, and it implements `ExtractLoadNodeProvider` and `LoadNodeProvider`?

**Comment by dockerzhang** [comment]
Please update related files if needed, for example:
https://inlong.apache.org/zh-CN/docs/next/design_and_concept/how_to_extend_data_node_for_manager

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/load/HudiProvider.java (hunk 1 above)
Comment (chestnut-c) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/load/HudiProvider.java]: > Can we merge the HudiProvider into one, and it implements `ExtractLoadNodeProvider` and `LoadNodeProvider`?


ok let me deal with it.

**Comment by chestnut-c** [comment]
> Please update related files if needed, for example: https://inlong.apache.org/zh-CN/docs/next/design_and_concept/how_to_extend_data_node_for_manager

Ok, I will modify the document after completion

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/StarRocksProvider.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public LoadNode createLoadNode(StreamNode nodeInfo, Map<String, StreamField> constantFieldMap) {
+        StarRocksSink starRocksSink = (StarRocksSink) nodeInfo;
+        Map<String, String> properties = parseProperties(starRocksSink.getProperties());
+        List<FieldInfo> fieldInfos = parseSinkFieldInfos(starRocksSink.getSinkFieldList(), starRocksSink.getSinkName());
+        List<FieldRelation> fieldRelations = parseSinkFields(starRocksSink.getSinkFieldList(), constantFieldMap);
+
+        Format format = null;
+        if (starRocksSink.getSinkMultipleEnable() != null && starRocksSink.getSinkMultipleEnable()
+                && StringUtils.isNotBlank(
+                        starRocksSink.getSinkMultipleFormat())) {
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/StarRocksProvider.java:59]:
```suggestion
        if (Boolean.TRUE.equals(starRocksSink.getSinkMultipleEnable())
                && StringUtils.isNotBlank(starRocksSink.getSinkMultipleFormat())) {
```
```

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/StreamType.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.common.consts;
+
+/**
+ * Constants of stream type is sink source public constant type.
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/StreamType.java:21]:
```suggestion
 * Constant for stream types, indicating that both StreamSource and StreamSink support these types.
```

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/LoadNodeProviderFactory.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+public class LoadNodeProviderFactory {
+
+    /**
+     * The load node provider collection
+     */
+    private static final List<LoadNodeProvider> LOAD_NODE_PROVIDER_LIST = new ArrayList<>();
+
+    static {
+        // The Providers Parsing SinkInfo to LoadNode which sort needed
+        LOAD_NODE_PROVIDER_LIST.add(new KafkaProvider());
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/LoadNodeProviderFactory.java:57]: Can we use 
```
@Autowired
    private List<LoadNodeProvider> LoadNodeProviderList;
```
?

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java [hunk 5]
```diff
@@ -19,30 +19,25 @@
...
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /**
- * The node provider factory.
+ * Factory of the extract node provider.
  */
-public class NodeProviderFactory {
+public class ExtractNodeProviderFactory {
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java:40]: Maybe should put this class from the `pojo` into the `service` directory.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/LoadNodeProviderFactory.java (hunk 4 above)
Comment (chestnut-c) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/LoadNodeProviderFactory.java:57]: > @Autowired

Public dependency manager-pojo is not well suited for @Autowired

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java (hunk 5 above)
Comment (chestnut-c) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java:40]: > Maybe should put this class from the `pojo` into the `service` directory.


The structure can be optimized in the next step, and the current changes should not be too large.

**Review by gong — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**